### PR TITLE
Deduplicate test_basic read_image

### PR DESCRIPTION
**Pull request by JohnKesapidesARM**
Merge readimage,readimage_int16,read_image_fp32,
readimage3d,readimage3d_int16,read_image3d_fp32
as they share a lot of common code.

Signed-off-by: John Kesapides <john.kesapides@arm.com>

### CODE REVIEW

**Comment by lakshmih** [comment]
Qualcomm is okay with this change

Sreelakshmi Haridas Maruthur
Qualcomm

**Review by lakshmih — APPROVED** [review]

**Comment by bashbaug** [comment]
Has one review and has had the focused review label for over a week, merging.

File: test_conformance/basic/test_readimage.cpp [hunk 1]
```diff
@@ -14,279 +14,364 @@
...
+                                       { CL_BGRA, CL_UNORM_INT8 } };
 
-        clReleaseKernel(kernel[i]);
-        clReleaseProgram(program[i]);
+    PASSIVE_REQUIRE_3D_IMAGE_SUPPORT(device)
+
+    int err = test_readimage<CL_MEM_OBJECT_IMAGE3D, cl_uchar>(
+        device, context, queue, &format[0]);
+
+    if (check_format(device, context, CL_MEM_OBJECT_IMAGE2D, format[1]))
```
Comment (jrprice) [inline test_conformance/basic/test_readimage.cpp:349]: @JohnKesapidesARM Presumably this should have been `CL_MEM_OBJECT_IMAGE2D` here?

Comment (JohnKesapidesARM) [reply test_conformance/basic/test_readimage.cpp:349]: yeap. I'll push a fix ASAP

**Pull request merged**